### PR TITLE
Update 01-introduction-to-high-dimensional-data.Rmd

### PR DESCRIPTION
**Pull request by hannesbecher**
<details>
<summary><strong>Instructions</strong></summary>

Thanks for contributing! :heart:

If this contribution is for instructor training, please email the link to this contribution to
checkout@carpentries.org so we can record your progress. You've completed your contribution
step for instructor checkout by submitting this contribution!

Keep in mind that **lesson maintainers are volunteers** and it may take them some time to
respond to your contribution. Although not all contributions can be incorporated into the lesson
materials, we appreciate your time and effort to improve the curriculum. If you have any questions
about the lesson maintenance process or would like to volunteer your time as a contribution
reviewer, please contact The Carpentries Team at team@carpentries.org.

You may delete these instructions from your comment.

\- The Carpentries
</details>

### CODE REVIEW

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 1]
```diff
@@ -37,19 +37,20 @@ knitr_fig_path("01-")
...
-in the data, $p$, are equal or larger than the number of observations (or data
-points), $n$. Unlike for *low-dimensional data* in which the number of observations,
-$n$, far outnumbers the number of features, $p$, analyses of high-dimensional data requires
-consideration of potential problems that come from having a large number of
-features.
+*High-dimensional data* are defined as data in which the number of features (variables observed),
+$p$, are close to or larger than the number of observations (or data points), $n$.
+The opposite is *low-dimensional data* in which the number of observations,
+$n$, far outnumbers the number of features, $p$. A related concept is *wide data*.
+This refers to data with numerous features irrespective of the number of observations.
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:44]:
```suggestion
efers to data with numerous features irrespective of the number of observations (similarly, *tall data* is often used to denote data with a large number of observations)
```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 2]
```diff
@@ -37,19 +37,20 @@ knitr_fig_path("01-")
...
+Analyses of high-dimensional data require consideration of potential problems that
+come from having a large number of features.
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:46]:
```suggestion
come from having more features than observations.

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 3]
```diff
@@ -90,7 +91,10 @@ knitr::include_graphics("../fig/intro-table.png")
 > 
 > > ## Solution
 > > 
-> > 2 and 4
+> > 1. No. The number of observations (100 patients) is far greater than the number of features (3).
+> > 2. Yes, this is an example of high-dimensional data. There are only 100 observations but 200,000+3 features.
+> > 3. No. There are many more observations than features.
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:96]:
```suggestion
> > 3. No. There are many more observations (200 patients) than features (5). 

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 4]
```diff
@@ -126,7 +130,8 @@ of the challenges we are facing when working with high-dimensional data.
 > ## Challenge 2 
 > 
 > Load the `Prostate` dataset from the `lasso2` package and examine the column
-> names. 
+> names. Although technically not a high-dimensional dataset, the `Prostate` data
+> will allow us explore the problems encountered when working with many features.
 >
 > Examine the dataset (in which each row represents a single patient) and plot
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:136]:
```suggestion
> Examine the dataset (in which each row represents a single patient) to:

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 5]
```diff
@@ -126,7 +130,8 @@ of the challenges we are facing when working with high-dimensional data.
 > ## Challenge 2 
 > 
 > Load the `Prostate` dataset from the `lasso2` package and examine the column
-> names. 
+> names. Although technically not a high-dimensional dataset, the `Prostate` data
+> will allow us explore the problems encountered when working with many features.
 >
 > Examine the dataset (in which each row represents a single patient) and plot
 > relationships between the variables using the `pairs` function. Why does it
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:137]:
```suggestion
> a) Determine how many observations ($n$) and features ($p$) are available (hint: see the `dim()` function) 
> b) Examine what variables were measured (hint: see the `names()` and `head()` functions)
> c) Plot the relationship between the variables (hint: see the `pairs()` function). 

```

Comment (catavallejos) [reply _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:137]: You will need to edit also the next line to add the 'Why does it' that was dropped from above.

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 6]
```diff
@@ -126,7 +130,8 @@ of the challenges we are facing when working with high-dimensional data.
 > ## Challenge 2 
 > 
 > Load the `Prostate` dataset from the `lasso2` package and examine the column
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:132]: Stop the sentence at `package'
```suggestion
> Load the `Prostate` dataset from the `lasso2` package.

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 7]
```diff
@@ -140,8 +145,12 @@ of the challenges we are facing when working with high-dimensional data.
...
 > > ```
 > > 
-> > ```{r view-prostate, eval = FALSE}
-> > View(Prostate)   #view the dataset
+> > ```{r dim-prostate, eval = FALSE}
+> > dim(Prostate)   #print the number of rows and columns
+> > ```
+> >
+> > ```{r head-prostate, eval = FALSE}
+> > head(Prostate)   #print the first 6 rows
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:153]:
```suggestion
>> names(Prostate) # examine the variable names
> > head(Prostate)   #print the first 6 rows

```

Comment (catavallejos) [reply _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:153]: also remove call to `names` in the next chunk

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 8]
```diff
@@ -154,20 +163,20 @@ of the challenges we are facing when working with high-dimensional data.
...
 > > 
 > {: .solution}
 {: .challenge}
 
 Imagine we are carrying out least squares regression on a dataset with 25
 observations. Fitting a best fit line through these data produces a plot shown
-in Figure 2a.
+in the left-hand panel of the figure below.
 
 However, imagine a situation in which the ratio of observations to features in
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:176]:
```suggestion
However, imagine a situation in which the number of observations and features in a dataset are almost equal.

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 9]
```diff
@@ -154,20 +163,20 @@ of the challenges we are facing when working with high-dimensional data.
...
 a dataset is almost equal. In that situation the effective number of
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:177]:
```suggestion
In that situation the effective number of

```

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd (hunk 8 above)
Comment (catavallejos) [reply _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:176]: This addresses Emma's comment in #39

File: _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd [hunk 10]
```diff
@@ -189,7 +198,7 @@ in these datasets makes high correlations between variables more likely.
 
 > ## Challenge 3
 > 
-> Use the `cor` function to examine correlations between all variables in the
+> Use the `cor()` function to examine correlations between all variables in the
```
Comment (catavallejos) [inline _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:201]: Remove lines 212 and 213 (` cor(Prostate[, c(1, 2, 4, 6, 9)])`). See Emma's review in #39

Comment (catavallejos) [reply _episodes_rmd/01-introduction-to-high-dimensional-data.Rmd:201]: also remove residual analysis

**Review by catavallejos — CHANGES_REQUESTED** [review]

**Review by hannesbecher — COMMENTED** [review]
Happy

**Comment by Alanocallaghan** [comment]
When teaching people, I tend to type `library("package")` because it's consistent between `library` and `install.packages` (or `BiocManager::install`, etc), and doesn't invite questions about non-standard evaluation that have complicated answers I am not very good at delivering.

**Comment by catavallejos** [comment]
> When teaching people, I tend to type `library("package")` because it's consistent between `library` and `install.packages` (or `BiocManager::install`, etc), and doesn't invite questions about non-standard evaluation that have complicated answers I am not very good at delivering.

I don't normally use quotes for this, but I take your argument. @hannesbecher can you please make sure that all library calls have the quotes? (for this and other episodes)

**Review by catavallejos — APPROVED** [review]

**Pull request merged**